utils/src/lib.rs

```diff
@@ -84,12 +85,13 @@ macro_rules! impl_to_fixed_bytes {
 /// - `wait_for_transaction_to_be_mined` is removed thanks to alloy's better builtin PendingTransaction await
 /// - DON'T use this if you want parse the exact revert reason/type, since this func will only give err msg like: "custom error 0x23b0db14",
 ///   instead, follow <https://docs.rs/alloy/0.12.5/alloy/contract/enum.Error.html#method.as_decoded_interface_error> to pattern-match err type
-pub async fn contract_send<T, P, C>(
-    call: &SolCallBuilder<T, P, C>,
+pub async fn contract_send<P, C, N>(
+    call: &SolCallBuilder<P, C, N>,
```

Comment:
we can simply drop this generic parameter `N` iiuc. because it has default to `N=Ethereum`, which is always what we need.

Reply:
removed in [baf9b3a](https://github.com/EspressoSystems/espresso-network/pull/3601/commits/baf9b3af6a6e3fe2c7109867b6da382a34dab90a)